### PR TITLE
Auto-format code at build time.

### PR DESCRIPTION
**Pull request by Naros**
https://issues.jboss.org/browse/DBZ-1392

This PR is based on the work in https://github.com/debezium/debezium/pull/1033.

### CODE REVIEW

**Comment by gunnarmorling** [comment]
@Naros, could you rebase this one, now that the other two code style PRs have been merged? Thanks!

**Comment by Naros** [comment]
@gunnarmorling Rebased.

**Comment by gunnarmorling** [comment]
Thanks! Now, if I were to apply the profile, will it go and do a gazillion of changes?

**Comment by Naros** [comment]
Yes, enabling the `format` profile should change the source.  The `validate-format` profile should just throw a build failure which is what I have enabled by default.

File: pom.xml [hunk 1]
```diff
@@ -485,6 +490,27 @@
...
+                    </configuration>
+                </plugin>
+                <plugin>
+                    <groupId>net.revelc.code</groupId>
+                    <artifactId>impsort-maven-plugin</artifactId>
+                    <version>${version.impsort}</version>
+                    <configuration>
+                        <groups>java.,javax.,org.,com.,io.</groups>
+                        <staticGroups>*</staticGroups>
+                        <staticAfter>true</staticAfter>
```
Comment (gunnarmorling) [inline pom.xml]:
```suggestion
                        <staticAfter>false</staticAfter>
```

I played a bit with this setting and it turns out that currently more files have static import before non-static ones (72 changed files vs. 162 changed files). So let's keep statics first.

Comment (gunnarmorling) [reply pom.xml]: So I pushed a change for this commit and also applying the import re-org actually.

**Comment by gunnarmorling** [comment]
I've also pushed one more commit for removing all trailing whitespace. All this to make the follow-up PR with the actual formatter application more digestable.

**Pull request merged**